docs/source/user_guide/pipeline-components.md

```diff
@@ -50,18 +50,186 @@ Component details and demo pipelines can be found in the `https://github.com/ely
+
+To list component registry entries:
+
+```bash
+$ elyra-metadata list component-registries
+
+Available metadata instances for component-registries (includes invalid):
+
+Schema               Instance                          Resource                                                                                                        
+------               --------                          --------                                                                                                        
+component-registry   elyra-airflow-filename-preconfig  .../jupyter/metadata/component-registries/elyra-airflow-filename-preconfig.json   
+```
+
+#### Adding components to the registry
+
+To add a component registry entry run `elyra-metadata install component-registries`.
+
+```bash
+$ elyra-metadata install component-registries \
+       --display_name="filter components" \
+       --description="filter text in files" \
+       --runtime=kfp \
+       --location_type=URL \
+       --paths="['https://raw.githubusercontent.com/elyra-ai/elyra/master/etc/config/components/kfp/filter_text_using_shell_and_grep.yaml']" \
+       --categories='["filter file content"]'
+```
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+#### Modifying a component registry entry
+
+To replace a component registry entry run `elyra-metadata install component-registries` and specify the `--replace` option:
+
+```bash
+$ elyra-metadata install component-registries \
+       --name="filter_components" \
+       --display_name="filter components" \
+       --description="filter text in files" \
+       --runtime=kfp \
+       --location_type=URL \
+       --paths="['https://raw.githubusercontent.com/elyra-ai/elyra/master/etc/config/components/kfp/filter_text_using_shell_and_grep.yaml']" \
+       --categories='["file operations"]' \
+       --replace
+```
+
+Note: You must specify all property values, not only the ones that you want to modify.
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+#### Deleting components from the registry
+
+To delete a component registry entry and its component definitions:
+
+```bash
+$ elyra-metadata remove component-registries \
+       --name="filter_components"
+```
+
+Refer to section [Configuration properties](#configuration-properties) for a parameter description.
+
+### Configuration properties
+
+The component registry entry properties are defined as follows. The string in the headings below, which is enclosed in parentheses, denotes the CLI option name.
+
+##### Name (display_name)
+
+A user-friendly name for the registry entry. Note that the registry entry name is not displayed in the palette. This property is required.
+
+Example: `data load components`
+
+##### N/A (name)
+
+A unique identifier for this registry entry. A value is generated from `Name` if no value is provided.
```

Comment:
"A unique identifier" sounds like a UUID.  I think something more like "The canonical name for this registry entry" might be slightly more correct.

Reply:
changed!